syntax = "proto3";

package hw1;

service ManageUserService {
  rpc RegisterUser (RegisterUserRequest) returns (UserActionResponse) {}
  rpc UpdateUser (UpdateUserRequest) returns (UserActionResponse) {}
  rpc DeleteUser (DeleteUserRequest) returns (UserActionResponse) {}
}

message RegisterUserRequest {
  string email = 1;
  string ticker = 2;
}

message UpdateUserRequest {
  string email = 1;
  string ticker = 2;
}

message DeleteUserRequest {
  string email = 1;
}

message UserActionResponse {
  string outcome = 1;
}

service StockService {
  rpc getLastStockValue (GetLastStockValueRequest) returns (GetLastStockValueResponse);
  rpc getStockPriceAverage (GetStockPriceAverageRequest) returns (GetStockPriceAverageResponse);
}

message GetLastStockValueRequest {
  string email = 1; 
}

message GetLastStockValueResponse {
  string ticker = 1; 
  float last_value = 2; 
  string timestamp = 3;  
}

message GetStockPriceAverageRequest {
  string email = 1; 
  int32 num_values = 2;  
}

message GetStockPriceAverageResponse {
  string ticker = 1;  
  float average_price = 2;  
  int32 num_values = 3;  
  string timestamp = 4;  
}
